### ISSUE TITLE
resource "azurerm_cosmosdb_cassandra_table" fails at upgrading the throughput

### ISSUE THREAD

**Issue by blame19** [hashicorp/terraform-provider-azurerm#12630]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

- Terraform 1.0.0
- azurerm (multiple versions, from 2.46 to 2.65)

### Affected Resource(s)

* `azurerm_cosmosdb_cassandra_table`

### Expected Behaviour

We were trying to resize the table autoscale_max_throughput from the minimum value of 4000 to anything else (5000, 8000).

### Actual Behaviour

Terraform produced a "Not Found" error:

```
[31m│[0m [0m[1m[31mError: [0m[0m[1mwaiting for update of Cassandra Table: (Table Name "test_table" / Cassandra Keyspace Name "dlgaas" / Database Account Name "matteocosmostest" / Resource Group "testcosmosdbmatt"): Code="NotFound" Message="Message: {\"code\":\"NotFound\",\"message\":\"Message: {\\\"Errors\\\":[\\\"Resource Not Found. Learn more: https:\\\\/\\\\/aka.ms\\\\/cosmosdb-tsg-not-found\\\"]}\\r\\nActivityId: 7026bcb6-7762-49e2-a1f4-d0e99959a705, Request URI: /apps/a6f25760-7394-4bfb-bafc-7ea08f3f3f13/services/66c05867-dd82-4754-86b2-868405aeb89f/partitions/c78a92f6-066e-42b3-9712-0618819f3022/replicas/132707812543386523s, RequestStats: \\r\\nRequestStartTime: 2021-07-16T15:19:33.2356989Z, RequestEndTime: 2021-07-16T15:19:33.2457029Z,  Number of regions attempted:1\\r\\nResponseTime: 2021-07-16T15:19:33.2457029Z, StoreResult: StorePhysicalAddress: rntbd://10.0.0.24:11300/apps/a6f25760-7394-4bfb-bafc-7ea08f3f3f13/services/66c05867-dd82-4754-86b2-868405aeb89f/partitions/c78a92f6-066e-42b3-9712-0618819f3022/replicas/132707812543386523s, LSN: 22, GlobalCommittedLsn: 22, PartitionKeyRangeId: , IsValid: True, StatusCode: 404, SubStatusCode: 0, RequestCharge: 1, ItemLSN: -1, SessionToken: -1#22, UsingLocalLSN: False, TransportException: null, BELatencyMs: 0.243, ActivityId: 7026bcb6-7762-49e2-a1f4-d0e99959a705, ResourceType: Database, OperationType: Read\\r\\nResponseTime: 2021-07-16T15:19:33.2457029Z, StoreResult: StorePhysicalAddress: rntbd://10.0.0.20:11300/apps/a6f25760-7394-4bfb-bafc-7ea08f3f3f13/services/66c05867-dd82-4754-86b2-868405aeb89f/partitions/c78a92f6-066e-42b3-9712-0618819f3022/replicas/132707812543386522s, LSN: 22, GlobalCommittedLsn: 22, PartitionKeyRangeId: , IsValid: True, StatusCode: 404, SubStatusCode: 0, RequestCharge: 1, ItemLSN: -1, SessionToken: -1#22, UsingLocalLSN: False, TransportException: null, BELatencyMs: 0.215, ActivityId: 7026bcb6-7762-49e2-a1f4-d0e99959a705, ResourceType: Database, OperationType: Read\\r\\n, SDK: Microsoft.Azure.Documents.Common/2.14.0\"}, Request URI: /dbs/matteocosmostest/colls, RequestStats: , SDK: Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0"[0m
...
Script execution finished with status: FAILURE
```

### Steps to Reproduce

1. Create a cassandra table with autoscale_settings.max_throughput = 4000
2. Plan & Apply
3. Edit the value to something greater
4. Plan
5. Try to apply

The operation succeeds when done via Azure Portal, so we can exclude that's an azure api issue.

**Comment by lorenzo-biava** [comment]
It kind of looks like the change is done, but TF fails when checking for the status update.

@manicminer can you take a quick look at this to help us understand if this issue can be caused by the Azure APIs more than by the provider itself?

```
INFO   2021-08-19 11:15:49    module.cdb.azurerm_cosmosdb_cassandra_table.example["test_sessions"]: Modifying... [id=/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-test-test-prod-cosmosdb/providers/Microsoft.DocumentDB/databaseAccounts/test-test-test-prod-cosmosdb/cassandraKeyspaces/test/tables/test_sessions]
INFO   2021-08-19 11:15:59    module.cdb.azurerm_cosmosdb_cassandra_table.example["test_sessions"]: Still modifying... [id=/subscriptions/d2e70fb0-f829-4d48-a9d9-...eyspaces/test/tables/test_sessions, 10s elapsed]
INFO   2021-08-19 11:16:09    module.cdb.azurerm_cosmosdb_cassandra_table.example["test_sessions"]: Still modifying... [id=/subscriptions/d2e70fb0-f829-4d48-a9d9-...eyspaces/test/tables/test_sessions, 20s elapsed]
INFO   2021-08-19 11:16:19    module.cdb.azurerm_cosmosdb_cassandra_table.example["test_sessions"]: Still modifying... [id=/subscriptions/d2e70fb0-f829-4d48-a9d9-...eyspaces/test/tables/test_sessions, 30s elapsed]
INFO   2021-08-19 11:16:22    
INFO   2021-08-19 11:16:22    Error: waiting for update of Cassandra Table: (Table Name "test_sessions" / Cassandra Keyspace Name "test" / Database Account Name "test-test-test-prod-cosmosdb" / Resource Group "test-test-test-prod-cosmosdb"): [...]
```

**Comment by blame19** [comment]
Adding last segment of the TRACE logs for the failure

```
2021-08-23T09:39:06.964Z [DEBUG] Azure Backend Response for https://nuancetfstatenonprod.blob.core.windows.net/xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate: 
HTTP/1.1 201 Created
Content-Length: 0
Content-Md5: GcEbiuLM0fzYEKAjwdO0fA==
Date: Mon, 23 Aug 2021 09:39:06 GMT
Etag: "0x8D96619DA237DB8"
Last-Modified: Mon, 23 Aug 2021 09:39:06 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Request-Id: 55a8df08-b01e-005d-2502-98ab79000000
X-Ms-Request-Server-Encrypted: true
X-Ms-Version: 2018-11-09
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mwaiting for update of Cassandra Table: (Table Name "test_table" / Cassandra Keyspace Name "dlgaas" / Database Account Name "matteocosmostest" / Resource Group "testcosmosdbmatt"): Code="NotFound" Message="Message: {\"code\":\"NotFound\",\"message\":\"Message: {\\\"Errors\\\":[\\\"Resource Not Found. Learn more: https:\\\\/\\\\/aka.ms\\\\/cosmosdb-tsg-not-found\\\"]}\\r\\nActivityId: e910935e-0a00-4600-be0c-d5bf6d72ee92, Request URI: /apps/7731b3c6-162b-4343-941b-b3338200d356/services/4fc459e4-2076-47cc-aae3-e91ff92da8e0/partitions/9c9bd1e8-a95a-4ad4-b037-e33dd1afe28c/replicas/132736776065509937s, RequestStats: \\r\\nRequestStartTime: 2021-08-23T09:38:41.5852410Z, RequestEndTime: 2021-08-23T09:38:41.5852410Z,  Number of regions attempted:1\\r\\nResponseTime: 2021-08-23T09:38:41.5852410Z, StoreResult: StorePhysicalAddress: rntbd://10.0.0.25:11300/apps/7731b3c6-162b-4343-941b-b3338200d356/services/4fc459e4-2076-47cc-aae3-e91ff92da8e0/partitions/9c9bd1e8-a95a-4ad4-b037-e33dd1afe28c/replicas/132736776065509937s, LSN: 13, GlobalCommittedLsn: 13, PartitionKeyRangeId: , IsValid: True, StatusCode: 404, SubStatusCode: 0, RequestCharge: 1, ItemLSN: -1, SessionToken: -1#13, UsingLocalLSN: False, TransportException: null, BELatencyMs: 0.365, ActivityId: e910935e-0a00-4600-be0c-d5bf6d72ee92, ResourceType: Database, OperationType: Read\\r\\nResponseTime: 2021-08-23T09:38:41.5852410Z, StoreResult: StorePhysicalAddress: rntbd://10.0.0.20:11000/apps/7731b3c6-162b-4343-941b-b3338200d356/services/4fc459e4-2076-47cc-aae3-e91ff92da8e0/partitions/9c9bd1e8-a95a-4ad4-b037-e33dd1afe28c/replicas/132736776065509938s, LSN: 13, GlobalCommittedLsn: 13, PartitionKeyRangeId: , IsValid: True, StatusCode: 404, SubStatusCode: 0, RequestCharge: 1, ItemLSN: -1, SessionToken: -1#13, UsingLocalLSN: False, TransportException: null, BELatencyMs: 0.506, ActivityId: e910935e-0a00-4600-be0c-d5bf6d72ee92, ResourceType: Database, OperationType: Read\\r\\n, SDK: Microsoft.Azure.Documents.Common/2.14.0\"}, Request URI: /dbs/matteocosmostest/colls, RequestStats: , SDK: Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0, Microsoft.Azure.Documents.Common/2.14.0"[0m
[31m│[0m [0m
[31m│[0m [0m[0m  with module.cdb.azurerm_cosmosdb_cassandra_table.example["test_table"],
[31m│[0m [0m  on .terraform/modules/cdb/main.tf line 91, in resource "azurerm_cosmosdb_cassandra_table" "example":
[31m│[0m [0m  91: resource "azurerm_cosmosdb_cassandra_table" "example" [4m{[0m[0m
[31m│[0m [0m
[31m╵[0m[0m
2021-08-23T09:39:06.969Z [DEBUG] Azure Backend Request: 
HEAD /xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate HTTP/1.1
Host: nuancetfstatenonprod.blob.core.windows.net
User-Agent: Terraform/1.0.1
X-Ms-Date: Mon, 23 Aug 2021 09:39:06 GMT
X-Ms-Lease-Id: c447ac4e-953e-0c87-e77c-664ae8add8da
X-Ms-Version: 2018-11-09
2021-08-23T09:39:06.990Z [DEBUG] Azure Backend Response for https://nuancetfstatenonprod.blob.core.windows.net/xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate: 
HTTP/1.1 200 OK
Content-Length: 24330
Accept-Ranges: bytes
Content-Md5: GcEbiuLM0fzYEKAjwdO0fA==
Content-Type: application/json
Date: Mon, 23 Aug 2021 09:39:06 GMT
Etag: "0x8D96619DA237DB8"
Last-Modified: Mon, 23 Aug 2021 09:39:06 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Access-Tier: Hot
X-Ms-Access-Tier-Inferred: true
X-Ms-Blob-Type: BlockBlob
X-Ms-Creation-Time: Mon, 23 Aug 2021 09:39:06 GMT
X-Ms-Lease-Duration: infinite
X-Ms-Lease-State: leased
X-Ms-Lease-Status: locked
X-Ms-Meta-Terraformlockid: eyJJRCI6ImM0NDdhYzRlLTk1M2UtMGM4Ny1lNzdjLTY2NGFlOGFkZDhkYSIsIk9wZXJhdGlvbiI6Ik9wZXJhdGlvblR5cGVBcHBseSIsIkluZm8iOiIiLCJXaG8iOiJyb290QGhhcm5lc3MtZGVsZWdhdGUtY2k0LXNmYnloby0yIiwiVmVyc2lvbiI6IjEuMC4xIiwiQ3JlYXRlZCI6IjIwMjEtMDgtMjNUMDk6Mzg6MzIuNjI4MDczMzY5WiIsIlBhdGgiOiJ4aW5mcmEtbm9ucHJvZC9zdGF0ZXMvaGFybmVzcy94aW5mcmEtdGVzdC9tYXR0ZW8tYWtzLW5vbi1wcm9kL3N0b3JhZ2UtdGVzdC12NS90ZXJyYWZvcm0udGZzdGF0ZSJ9
X-Ms-Request-Id: 55a8df11-b01e-005d-2e02-98ab79000000
X-Ms-Server-Encrypted: true
X-Ms-Version: 2018-11-09
2021-08-23T09:39:06.990Z [DEBUG] Azure Backend Request: 
HEAD /xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate HTTP/1.1
Host: nuancetfstatenonprod.blob.core.windows.net
User-Agent: Terraform/1.0.1
X-Ms-Date: Mon, 23 Aug 2021 09:39:06 GMT
X-Ms-Lease-Id: c447ac4e-953e-0c87-e77c-664ae8add8da
X-Ms-Version: 2018-11-09
2021-08-23T09:39:06.994Z [DEBUG] Azure Backend Response for https://nuancetfstatenonprod.blob.core.windows.net/xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate: 
HTTP/1.1 200 OK
Content-Length: 24330
Accept-Ranges: bytes
Content-Md5: GcEbiuLM0fzYEKAjwdO0fA==
Content-Type: application/json
Date: Mon, 23 Aug 2021 09:39:06 GMT
Etag: "0x8D96619DA237DB8"
Last-Modified: Mon, 23 Aug 2021 09:39:06 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Access-Tier: Hot
X-Ms-Access-Tier-Inferred: true
X-Ms-Blob-Type: BlockBlob
X-Ms-Creation-Time: Mon, 23 Aug 2021 09:39:06 GMT
X-Ms-Lease-Duration: infinite
X-Ms-Lease-State: leased
X-Ms-Lease-Status: locked
X-Ms-Meta-Terraformlockid: eyJJRCI6ImM0NDdhYzRlLTk1M2UtMGM4Ny1lNzdjLTY2NGFlOGFkZDhkYSIsIk9wZXJhdGlvbiI6Ik9wZXJhdGlvblR5cGVBcHBseSIsIkluZm8iOiIiLCJXaG8iOiJyb290QGhhcm5lc3MtZGVsZWdhdGUtY2k0LXNmYnloby0yIiwiVmVyc2lvbiI6IjEuMC4xIiwiQ3JlYXRlZCI6IjIwMjEtMDgtMjNUMDk6Mzg6MzIuNjI4MDczMzY5WiIsIlBhdGgiOiJ4aW5mcmEtbm9ucHJvZC9zdGF0ZXMvaGFybmVzcy94aW5mcmEtdGVzdC9tYXR0ZW8tYWtzLW5vbi1wcm9kL3N0b3JhZ2UtdGVzdC12NS90ZXJyYWZvcm0udGZzdGF0ZSJ9
X-Ms-Request-Id: 55a8df22-b01e-005d-3e02-98ab79000000
X-Ms-Server-Encrypted: true
X-Ms-Version: 2018-11-09
2021-08-23T09:39:06.994Z [DEBUG] Azure Backend Request: 
PUT /xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate?comp=metadata HTTP/1.1
Host: nuancetfstatenonprod.blob.core.windows.net
User-Agent: Terraform/1.0.1
Content-Length: 0
X-Ms-Date: Mon, 23 Aug 2021 09:39:06 GMT
X-Ms-Lease-Id: c447ac4e-953e-0c87-e77c-664ae8add8da
X-Ms-Version: 2018-11-09
Accept-Encoding: gzip
2021-08-23T09:39:07.002Z [DEBUG] Azure Backend Response for https://nuancetfstatenonprod.blob.core.windows.net/xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate?comp=metadata: 
HTTP/1.1 200 OK
Content-Length: 0
Date: Mon, 23 Aug 2021 09:39:06 GMT
Etag: "0x8D96619DA28FCF3"
Last-Modified: Mon, 23 Aug 2021 09:39:06 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Request-Id: 55a8df2c-b01e-005d-4702-98ab79000000
X-Ms-Request-Server-Encrypted: true
X-Ms-Version: 2018-11-09
2021-08-23T09:39:07.002Z [DEBUG] Azure Backend Request: 
PUT /xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate?comp=lease HTTP/1.1
Host: nuancetfstatenonprod.blob.core.windows.net
User-Agent: Terraform/1.0.1
Content-Length: 0
X-Ms-Date: Mon, 23 Aug 2021 09:39:07 GMT
X-Ms-Lease-Action: release
X-Ms-Lease-Id: c447ac4e-953e-0c87-e77c-664ae8add8da
X-Ms-Version: 2018-11-09
Accept-Encoding: gzip
2021-08-23T09:39:07.007Z [DEBUG] Azure Backend Response for https://nuancetfstatenonprod.blob.core.windows.net/xinfra-nonprod/states%2Fharness%2Fxinfra-test%2Fmatteo-aks-non-prod%2Fstorage-test-v5%2Fterraform.tfstate?comp=lease: 
HTTP/1.1 200 OK
Content-Length: 0
Date: Mon, 23 Aug 2021 09:39:06 GMT
Etag: "0x8D96619DA28FCF3"
Last-Modified: Mon, 23 Aug 2021 09:39:06 GMT
Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
X-Ms-Request-Id: 55a8df40-b01e-005d-5602-98ab79000000
X-Ms-Version: 2018-11-09
2021-08-23T09:39:07.008Z [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021-08-23T09:39:07.014Z [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/azurerm/2.67.0/linux_amd64/terraform-provider-azurerm_v2.67.0_x5 pid=11370
2021-08-23T09:39:07.014Z [DEBUG] provider: plugin exited
Script execution finished with status: FAILURE
```

**Comment by lorenzo-biava** [comment]
Looks like the issue is with an argument being wrong here: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/cosmos/cosmosdb_cassandra_table_resource.go#L182

**Comment by tombuildsstuff** [comment]
fixed via  #13102
